### PR TITLE
Implement Clock, Diary Sexp and Planning

### PR DESCRIPTION
**Pull request by schoettl**
- [x] Update examples in README, but not before #40 is merged

### CODE REVIEW

File: test/org_parser/parser_test.cljc [hunk 1]
```diff
@@ -1004,3 +1029,60 @@ is another section"))))))
...
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:46"]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse "CLOCK: [2021-05-22 Sat 23:26]--[2021-05-22 Sat 23:46] =>  0:20"))))
+    (testing "a simple clock line"
+      (is (= [:clock [:timestamp-inactive-range [:ts-inner-span
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]]
+               [:ts-time "23:46"]
+               [:ts-modifiers]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse " CLOCK: [2021-05-22 Sat 23:26-23:46] =>  0:20 "))))
```
Comment (munen) [inline test/org_parser/parser_test.cljc:1048]: @schoettl It's pretty cool that the parser can read the shorthand. Can Emacs org mode, though? It doesn't seem to work for me and I also haven't found documentation for Org mode that shows this syntax for `CLOCK` entries.

I mean, if the parser can do it 'by accident', then that's totally fine, of course. However, I'm curious if it's part of the standard and where that would be documented.

File: test/org_parser/parser_test.cljc [hunk 2]
```diff
@@ -1004,3 +1029,60 @@ is another section"))))))
...
+    ))
+
+
+(deftest planning
+  (let [parse #(parser/org % :start :planning)]
+    (testing "a simple planning info"
+      (is (= [:planning [:planning-info
+                         [:planning-keyword [:planning-kw-scheduled]]
+                         [:timestamp [:timestamp-inactive [:ts-inner [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]] [:ts-modifiers]]]]]]
+             (parse "SCHEDULED: [2021-05-22 Sat 23:26]"))))
```
Comment (munen) [inline test/org_parser/parser_test.cljc:1070]: Does it make sense to have inactive timestamps as a planning item? It will not show up in the agenda like this.

I've quickly tried to look up if there's a flag to configure it like this and couldn't find it.

**Review by munen — APPROVED** [review]
@schoettl Generally speaking, this looks good to me. I have a couple comprehension questions and added them inline as comments. But they are not a blocker for merging. Good work :+1:

File: test/org_parser/parser_test.cljc (hunk 2 above)
Comment (schoettl) [reply test/org_parser/parser_test.cljc:1070]: Thanks for the feedback :) I tried to stick to the worg spec. It says, the planning info has to contain a timestamp and links to the timestamps. It doesn't say, that it must be an active one. In other places the spec distinguishes between active and inactive, e.g. at the `CLOCK:` entries.

I look at inactive timestamps in planning information as "temporary disabled".

File: test/org_parser/parser_test.cljc (hunk 1 above)
Comment (schoettl) [reply test/org_parser/parser_test.cljc:1048]: The worg spec says, there must be an inactive timestamp range after `CLOCK: `. Timestamps are defined in the spec as:
```
...
[DATE TIME REPEATER-OR-DELAY]--[DATE TIME REPEATER-OR-DELAY]   (inactive range)
[DATE TIME-TIME REPEATER-OR-DELAY]                             (inactive range)
```

So I think I stick to the worg spec. But what is the real spec? Org mode or the [worg spec](https://orgmode.org/worg/dev/org-syntax.html#Timestamp)? ;) Our org-parser README says, the worg spec leads to the reference implementation. It might also be vice versa or so so ^^

Comment (schoettl) [reply test/org_parser/parser_test.cljc:1048]: In any case, I think it's good to allow all possible timestamp ranges here.

File: test/org_parser/parser_test.cljc (hunk 2 above)
Comment (munen) [reply test/org_parser/parser_test.cljc:1070]: Parsing them is certainly not wrong in case somebody does it, so it's pretty cool that your implementation can already do it(;

However, I haven't seen it so far and wouldn't see a reason to do it this way. The [docs on schedules and deadline](https://orgmode.org/manual/Deadlines-and-Scheduling.html) also show examples with active timestamps - albeit without explicitly saying that this is the only option(;

Comment (schoettl) [reply test/org_parser/parser_test.cljc:1070]: It's true that only active planning info must appear in the agenda. We have to consider that later. I personally like the option to disable planning info without deleting the planning info, e.g. a cancelled event.

We should discuss anyway, how strict the parser should be. We can talk about this next weekend :)

File: test/org_parser/parser_test.cljc (hunk 1 above)
Comment (munen) [reply test/org_parser/parser_test.cljc:1048]: Well, the worg 'spec' is a very traditional document driven 'spec'. The canonical implementation of Org mode is org-element.el which links to 'see more details about Org' (https://github.com/emacs-mirror/emacs/blob/fe15532eef5e7375fd2d91b92e173e5ed12534ef/lisp/org/org-element.el#L25).

It's as always - as long as the documentation is not automatically generated, or even better - the code generated from the documentation, then there's always a good chance of discrepancy. That's why having a BNF based parser is quite superior in the long run: Less chance of ambiguities between spec and implementation(;  

In any case, Emacs doesn't have the regular `CLOCK` functionality for the shorthand syntax. If I update the times and `C-c C-c` it, there's no update in the calculated time. However, it does certainly parse it correctly for the syntax highlighting, at least(;

TL;DR: Parsing more is never wrong, so it's great that org-parser does it! But I don't think it's valid syntax according to what actually works in Emacs.

Comment (schoettl) [reply test/org_parser/parser_test.cljc:1048]: Alright, we will definitely export/render the timestamps `[]--[]` which Org mode understands :)

Comment (munen) [reply test/org_parser/parser_test.cljc:1048]: :+1:
